### PR TITLE
Add docs for Oblivious DNS over HTTPS

### PR DESCRIPTION
**Pull request by tanyav2**
cc @Lekensteyn @chris-wood @grittygrease

### CODE REVIEW

**Review by Lekensteyn — CHANGES_REQUESTED** [review]
This should not just be a statement of work that we have done. It should be rewritten to be more user-centric:

- How can can a user use ODoH?
- How can someone set up a proxy or target?
- Which draft version is implemented?

It should emphasize the this is experimental, users should not rely on it for critical purposes.

File: products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+order: 7
+---
+
+# Oblivious DNS over HTTPS
+
+Oblivious DNS over HTTPS (or ODoH) is a new DNS protocol still in the works that separates client IP addresses from their DNS queries by routing the DNS queries over a proxy, so that the DNS resolver cannot link client IPs with their queries. You can learn more about it in this [blog post](https://blog.cloudflare.com/oblivious-dns/). Please note that this protocol and all related implementations are currently **experimental**, so use at your own risk.
```
Comment (abracchi-tw) [inline products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md:7]: Please note that this protocol and all related implementations are currently **experimental**.

File: products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Follow the progress of ODoH by following the IETF draft [here](https://datatracker.ietf.org/doc/draft-pauly-dprive-oblivious-doh/).
```
Comment (abracchi-tw) [inline products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md:9]: To follow the progress of ODoH, check out this [IETF draft](https://datatracker.ietf.org/doc/draft-pauly-dprive-oblivious-doh/).

File: products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+We have open sourced our interoperable ODoH library implementations in Rust, [odoh-rs](https://github.com/cloudflare/odoh-rs), and Go, [odoh-go](https://github.com/cloudflare/odoh-go), which you can use to implement your own client or target resolver.
```
Comment (abracchi-tw) [inline products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md:11]: You can use our open-source, interoperable ODoH library implementation (both in [Rust](https://github.com/cloudflare/odoh-rs) and [Go](https://github.com/cloudflare/odoh-go)) to implement your own client or target resolver. Our test clients are also open source in both [Rust](https://github.com/cloudflare/odoh-client-rs) and [Go](https://github.com/cloudflare/odoh-client-go), which you can use to demo ODoH queries.

File: products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+We have open sourced our interoperable ODoH library implementations in Rust, [odoh-rs](https://github.com/cloudflare/odoh-rs), and Go, [odoh-go](https://github.com/cloudflare/odoh-go), which you can use to implement your own client or target resolver. 
+
+We have also open sourced test clients in Rust, [odoh-client-rs](https://github.com/cloudflare/odoh-client-rs) and Go, [odoh-client-go](https://github.com/cloudflare/odoh-client-go), which can be used to demo ODoH queries. 
+
+We have also open sourced [odoh-server-go](https://github.com/cloudflare/odoh-server-go) as a sample server that can be used to run a proxy or a target for ODoH. `odoh-server-go` can act as both a proxy and a target. It listens on `/proxy` for proxy requests, and `/dns-query` for target requests. You could even run two instances of it and have one instance function as a proxy and the other as a target.
+
+Finally, we have implemented support for ODoH queries in 1.1.1.1.
+
+We are not aware of any DNS clients like [dnscrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy) that currently implement ODoH, but watch out for developments in the future!
```
Comment (abracchi-tw) [inline products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md:19]: do we need to specify this? can we re-word it to just say:

For the time being, DNS clients like [dnscrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy) don't implement ODoH.

File: products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## Setting up proxy and target
+
+Here's an example that demonstrates running your own proxy, and using 1.1.1.1 as a target.
+
+In one terminal, start the proxy:
```
Comment (abracchi-tw) [inline products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md:25]: 1. In a Terminal window, start the proxy:

File: products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+In one terminal, start the proxy: 
+
+```bash
+$ git clone https://github.com/cloudflare/odoh-server-go
+$ cd odoh-server-go/
+$ go build
+$ PORT=8080 ./odoh-server-go
+```
+
+From another terminal, use the client to send queries:
```
Comment (abracchi-tw) [inline products/1.1.1.1/src/content/fun-stuff/oblivious-dns-over-https.md:34]: 2. Open a second Terminal window and use the client to send queries:

**Review by abracchi-tw — CHANGES_REQUESTED** [review]

**Comment by abracchi-tw** [comment]
hey @Lekensteyn I see there are some changes still pending - happy to chat about them if needed, but I'd like to close out/merge this one!

**Review by 888520ll19811209ll — APPROVED** [review]

**Comment by marciocloudflare** [comment]
Closing as this is outdated

**Pull request closed without merging**